source/cellular_pkthandler.c

```diff
@@ -94,117 +90,104 @@ static CellularPktStatus_t _convertAndQueueRespPacket( CellularContext_t * pCont
-
-    /* pInputLine = "+" pTokenPtr + ":" + pSavePtr.
-     * if string not start with "+", then pTokenPtr = pSavePtr = pInputPtr. */
-    char * pSavePtr = pInputLine, * pTokenPtr = pInputLine;
-
-    LogDebug( ( "Next URC token to parse [%s]", pInputLine ) );
-
-    /* Check if prefix exist in the input string. The pInputLine is checked in _processUrcPacket. */
-    ( void ) Cellular_ATIsPrefixPresent( pInputLine, &inputWithPrefix );
-
-    if( inputWithPrefix == true )
-    {
-        /* Cellular_ATIsPrefixPresent check the prefix string is valid and start with
-         * leading char. ":" is also checked in Cellular_ATIsPrefixPresent. Remove
-         * the leading char and split the string. */
-        pSavePtr++;
-        pTokenPtr = strtok_r( pSavePtr, ":", &pSavePtr );
-
-        if( pTokenPtr == NULL )
-        {
-            LogError( ( "_Cellular_AtParse : input string error, start with \"+\" but no token %s", pInputLine ) );
-            pktStatus = CELLULAR_PKT_STATUS_BAD_REQUEST;
-        }
-    }
-    else
+    if( ( pContext->pktRespCB != NULL ) && ( pktStatus == CELLULAR_PKT_STATUS_OK ) )
     {
-        /* This is the input without prefix case. Nothing need to be done for this case.
-         * There are some special cases. For example, "+URC" the string without  ":" should
-         * be regarded as URC without prefix. */
+        pktStatus = pContext->pktRespCB( pContext,
+                                         ( const CellularATCommandResponse_t * ) pBuf,
+                                         pContext->pPktUsrData,
+                                         pContext->PktUsrDataLen );
     }
 
-    if( pktStatus == CELLULAR_PKT_STATUS_OK )
+    /* Notify calling thread, Not blocking immediately comes back if the queue is full. */
+    if( xQueueSend( pContext->pktRespQueue, ( void * ) &pktStatus, ( TickType_t ) 0 ) != pdPASS )
     {
-        /* Now get the handler function based on the token. */
-        pktStatus = _atParseGetHandler( pContext, pTokenPtr, pSavePtr );
+        pktStatus = CELLULAR_PKT_STATUS_FAILURE;
+        LogError( ( "_convertAndQueueRespPacket: Got a response when the Resp Q is full!!" ) );
     }
 
     return pktStatus;
 }
 
 /*-----------------------------------------------------------*/
 
-/**
- * @brief copy the URC log in the buffer to a heap memory and process it.
- */
 static CellularPktStatus_t _processUrcPacket( CellularContext_t * pContext,
                                               const char * pBuf )
 {
     CellularPktStatus_t pktStatus = CELLULAR_PKT_STATUS_OK;
+    bool inputWithPrefix = false;
+    char * pInputLine = NULL;
+    char * pSavePtr = NULL, * pTokenPtr = NULL;
     CellularATError_t atStatus = CELLULAR_AT_SUCCESS;
-    char * payload = NULL;
 
-    if( pBuf != NULL )
+    LogDebug( ( "Next URC token to parse [%s]", pInputLine ) );
+
+    /* pBuf is checked in _Cellular_HandlePacket. */
+    atStatus = Cellular_ATStrDup( &pInputLine, pBuf );
```

Comment:
```suggestion
    /* pBuf is checked in _Cellular_HandlePacket. */
    atStatus = Cellular_ATStrDup( &pInputLine, pBuf );

    LogDebug( ( "Next URC token to parse [%s]", pInputLine ) );
```
I think you should copy it before printing.